backend/oasst_backend/prompt_repository.py

```diff
@@ -1069,6 +1069,39 @@ def mark_messages_deleted(self, messages: Message | UUID | list[Message | UUID],
 
                 parent_ids = self.db.execute(query).scalars().all()
 
+    @managed_tx_method(CommitMode.COMMIT)
+    def reintroduce_deleted_message(self, message: Message | UUID, recursive: bool = True):
+        """
+        Reintroduce deleted messages and all their descendants.
+        """
+
+        message_id = None
+        if isinstance(message, UUID):
+            message_id = message
+        elif isinstance(message, Message):
+            message_id = message.id
+        else:
+            raise OasstError("Server error", OasstErrorCode.SERVER_ERROR1, HTTPStatus.INTERNAL_SERVER_ERROR)
+
+        query = update(Message).where(Message.id == message_id).values(deleted=False)
+        self.db.execute(query)
+
+        if recursive:
+            parent_id = None
+            if isinstance(message, UUID):
+                parent_id = self.db.query(Message.parent_id).where(Message.id == message).first()[0]
```

Comment:
`Message.id == message` looks a bit strange, should it be `message_id`? Do you want to undelete all descendants? The easiest would be to use the `fetch_message_descendants()` method and use ORM to set deleted = False .. usually message trees have <100 messages.

Reply:
yes it could totally be messag_id, don't know why i did not do that.

I don't want to undelete every descendant only the parents like 
A (deleted)
|->A.b (deleted)
|-> A.c (deleted)

I undelete A.c so now the tree mus look like this : 

A (not deleted)
|->A.b (deleted)
|-> A.c (not deleted)

A.b stay deleted because it is not a parent of A.c

Reply:
oh, undeleting parents is indeed a VERY important and should not be optional - otherwise a broken tree is created which is bad - especially if they are still active (will show broken-conversation errors to users and will require manual fixes of the DB). If the recursive parameter was for the trace to root of the tree please do do not expose it to the REST API (and maybe could we rename it to undelete_parents)  ... it must not be possible to undelete an isolated child of a deleted parent via API.

Reply:
That's what I thought, I will remove the recursive option and make that `undelete` always check if the parent is deleted or not, and if it's deleted proceed to `undelete` the parent (doing this recursively ofc)

I'm not sure to rename the function `undelete_parent` just keep `undelete` and change the description to say that it `undelete` parents recursively :)